types/factory-girl/index.d.ts

```diff
@@ -12,11 +12,20 @@ declare const factory: factory.Static;
 declare namespace factory {
     type Generator<T> = () => T;
 
-    type Definition<T> = T | Generator<T>;
+    type Definition<T> = T | Generator<T> | Promise<T>;
 
-    type Attributes<T> = Definition<{
+    type Attributes<T> = {
         [P in keyof T]: Definition<T[P]>;
-    }>;
+    };
+
+    type MaybeReadonlyArray<T> = T | ReadonlyArray<T>;
```

Comment:
Most of `Factory#...Many()` function can take `T` or `ReadonlyArray<T>` for its attributes and options. So I defined a utility type for it.

- https://github.com/simonexmachina/factory-girl/blob/f7ec9d5004021ff1334ab9440e9b5cc4fd1a18b0/src/Factory.js#L69